### ISSUE TITLE
store savegames in their own folder

### ISSUE THREAD

**Issue by bmorel** [wesnoth/wesnoth#7536]
### Describe the desired feature

Currently, when someone wants to a load a game, they have to browse the whole list of saves, from all games mixed together.

My idea is to store all savegames related to a unique game into a folder on disk. This would allow the savegame menu selector to fold/unfold games to make it easier to load a specific game. Perhaps it would also be possible to only load the most recent savegame of a file when it is "folded".

Hopefully my description is clear enough? Also, I talk about folders because it's what comes to my mind, but I think wesnoth already have a mechanism like that, since it is able to "go back to turn N" in solo games.

**Comment by cooljeanius** [comment]
I think there's another issue about this, but I can't find it at the moment... (it's something I've wanted, too)

**Comment by bmorel** [comment]
I quickly tried to find another issue, since I thought someone should have had it too, but I admit I gone lazy when I noticed there was several pages of issues: mea culpa

**Comment by Wedge009** [comment]
I don't recall this precise issue being logged before so no need to worry about duplication here. Some of the others aren't as diligent in avoiding duplication or consolidating similar issues, so we have an excess of issues logged. I recognise it's not friendly to newcomers.

**Comment by demario-w** [comment]
#1089

**Comment by Wedge009** [comment]
If I understand correctly, that one is about segregating automatic and manual save games, this one is about separating by some other grouping. But they are quite similar - could this one be merged into it?

**Comment by stevecotton** [comment]
Is this about grouping by campaign, as in "they have to browse the whole list of saves, from all ***campaigns*** mixed together."?

There's #1317 about adding a "load" or "continue" button to the campaign menu, which would either load the most recent save of that campaign, or open the load dialog with a filter set up to only show files from that campaign. That might also have something in #4543.

There's some overlap with #5118, which is about branching campaigns where a choice can lead to different scenarios or even a different ending.

**Comment by bmorel** [comment]
> Is this about grouping by campaign, as in "they have to browse the whole list of saves, from all campaigns mixed together."?

No, it not just campaigns, it's all games, including multiplayer ones.

**Comment by soliton-** [comment]
It would be good to explain what a unique game is.

**Comment by gfgtdf** [comment]
I'd read the first post as each started scenario is its own unique game. But it could also be interpreted as per campaign i guess

**Comment by bmorel** [comment]
> But it could also be interpreted as per campaign i guess

That's what I had in mind. Although this would be a bit more complicated for MP scenarios or campaigns: then, I guess names of players plus the map/campaign name plus a date would be needed to identify a "game"?
